src/proxy/List.ts

```diff
@@ -0,0 +1,148 @@
+        return this.encodeInvoke<boolean>(ListCompareAndRemoveAllCodec, this.serializeList(elements));
+    }
+
+    retainAll(elements: E[]): Q.Promise<boolean> {
+        return this.encodeInvoke<boolean>(ListCompareAndRetainAllCodec, this.serializeList(elements));
+    }
+
+    removeAt(index: number): Q.Promise<E> {
+        return this.encodeInvoke<E>(ListRemoveWithIndexCodec, index);
+    }
+
+    get(index: number): Q.Promise<E> {
+        return this.encodeInvoke<E>(ListGetCodec, index);
+    }
+
+    set(index: number, element: E): Q.Promise<E> {
+        return this.encodeInvoke<E>(ListSetCodec, index, this.toData(element));
+    }
+
+    indexOf(element: E): Q.Promise<number> {
+        return this.encodeInvoke<number>(ListIndexOfCodec, this.toData(element));
+    }
+
+    lastIndexOf(element: E): Q.Promise<number> {
+        return this.encodeInvoke<number>(ListLastIndexOfCodec, this.toData(element));
+    }
+
+    size(): Q.Promise<number> {
+        return this.encodeInvoke<number>(ListSizeCodec);
+    }
+
+    subList(start: number, end: number): Q.Promise<E[]> {
+        return this.encodeInvoke(ListSubCodec, start, end).then((encoded: Data[]) => {
+            return encoded.map((item: Data) => {
+                return this.toObject(item);
+            });
+        });
+    }
+
+    toArray(): Q.Promise<E[]> {
+        return this.encodeInvoke(ListGetAllCodec)
+            .then((elements: Array<Data>) => {
+                return elements.map((element) => {
+                    return this.toObject(element);
+                });
+            });
+    }
+
+    addItemListener(listener: ItemListener<E>, includeValue: boolean): Q.Promise<string> {
+        var request = ListAddListenerCodec.encodeRequest(this.name, includeValue, false);
+        var handler = (message: ClientMessage) => {
+            ListAddListenerCodec.handle(message, (element: Data, uuid: string, eventType: number) => {
+                var responseObject = element ? this.toObject(element) : null;
+                var listenerFunction: Function;
+                if (eventType === 1) {
+                    listenerFunction = listener.itemAdded;
+                } else if (eventType === 2) {
+                    listenerFunction = listener.itemRemoved;
+                }
+
+                if (listenerFunction) {
+                    listenerFunction.call(responseObject, responseObject);
+                }
+            });
+        };
+        return this.client.getListenerService().registerListener(request, handler,
+            ListAddListenerCodec.decodeResponse, this.name);
+    }
+
+    removeItemListener(registrationId: string): Q.Promise<boolean> {
+        return this.client.getListenerService().deregisterListener(
+            ListRemoveListenerCodec.encodeRequest(this.name, registrationId),
+            ListRemoveListenerCodec.decodeResponse
+        );
+    }
+
+    private serializeList(input: Array<any>): Array<Data> {
```

Comment:
I think serializeList can take only `Array<E>`. There is no need to allow `Array<any>`.